packages/itwin/property-grid/src/test/components/PropertyGridContent.test.tsx

```diff
@@ -217,4 +221,218 @@ describe("<PropertyGridContent />", () => {
+        <TelemetryContextProvider onFeatureUsed={onFeatureUsedSpy}>
+          <PropertyGridContent dataProvider={provider} imodel={imodel} settingsMenuItems={[(props) => <ShowHideNullValuesSettingsMenuItem {...props} />]} />
+        </TelemetryContextProvider>,
+      );
+
+      const settingsButton = await waitFor(() => getByRole("button", { name: "settings.label" }));
+      await user.click(settingsButton);
+
+      const setting = await waitFor(() => getByText("settings.hide-null.label"));
+      await user.click(setting);
+
+      await waitFor(() => {
+        expect(queryByText("Test Prop")).to.not.be.null;
+        expect(queryByText("Null Prop")).to.be.null;
+      });
+
+      // Wait for throttling to timeout
+      await new Promise((r) => setTimeout(r, 1000));
+      onFeatureUsedSpy.resetHistory();
+      act(() => provider.onDataChanged.raiseEvent());
+
+      await waitFor(() => {
+        expect(onFeatureUsedSpy).to.be.calledWith("hide-empty-values-enabled");
+        expect(onFeatureUsedSpy).to.not.be.calledWith("hide-empty-values-disabled");
+        expect(onFeatureUsedSpy).to.not.be.calledWith("filter-properties");
+      });
+    });
+
+    it("does not report when data changes and no values are loaded", async () => {
+      const imodel = {} as IModelConnection;
+      const onFeatureUsedSpy = sinon.spy();
+      const dataProvider = {
+        onDataChanged: new BeEvent(),
+        getData: createFunctionStub<PresentationPropertyDataProvider["getData"]>(),
+        keys: new KeySet([{ className: "class", id: "id" }]),
+      };
+
+      dataProvider.getData.resolves({
+        categories: [],
+        records: {},
+        label: PropertyRecord.fromString("Test Label"),
+        description: "TestClassName",
+      });
+
+      const { user, queryByText, getByRole, getByText, rerender } = renderWithContext(
+        <TelemetryContextProvider onFeatureUsed={onFeatureUsedSpy}>
+          <PropertyGridContent
+            dataProvider={dataProvider as unknown as IPresentationPropertyDataProvider}
+            imodel={imodel}
+            settingsMenuItems={[(props) => <ShowHideNullValuesSettingsMenuItem {...props} />]}
+          />
+        </TelemetryContextProvider>,
+      );
+
+      const settingsButton = await waitFor(() => getByRole("button", { name: "settings.label" }));
+      await user.click(settingsButton);
+
+      const setting = await waitFor(() => getByText("settings.hide-null.label"));
+      await user.click(setting);
+
+      await waitFor(() => {
+        expect(queryByText("Test Label")).to.not.be.null;
+      });
+
+      onFeatureUsedSpy.resetHistory();
+      dataProvider.keys = new KeySet();
+      dataProvider.getData.reset();
+      dataProvider.getData.resolves({
+        categories: [],
+        records: {},
+        label: PropertyRecord.fromString("New Test Label"),
+        description: "NewTestClassName",
+      });
+      act(() => provider.onDataChanged.raiseEvent());
+
+      rerender(
```

Comment:
Why is this `rerender` is needed? Looks like `PropertyGridContent` is rerendered with same props.

Reply:
Without the `rerender` it would not update the displayed data. Removed the test as it no longer is relevant.